nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateTableFetch.java

```diff
@@ -0,0 +1,254 @@
+                int rowCount = resultSet.getInt(1);
+
+                // Update the state map with the newly-observed maximum values
+                ResultSetMetaData rsmd = resultSet.getMetaData();
+                for (int i = 2; i <= rsmd.getColumnCount(); i++) {
+                    String resultColumnName = rsmd.getColumnName(i).toLowerCase();
+                    int type = rsmd.getColumnType(i);
+                    try {
+                        String newMaxValue = getMaxValueFromRow(resultSet, i, type, statePropertyMap.get(resultColumnName.toLowerCase()), dbAdapter.getName());
+                        if (newMaxValue != null) {
+                            statePropertyMap.put(resultColumnName, newMaxValue);
+                        }
+                    } catch (ParseException | IOException pie) {
+                        // Fail the whole thing here before we start creating flow files and such
+                        throw new ProcessException(pie);
+                    }
+                }
+
+                final int numberOfFetches = (partitionSize == 0) ? rowCount : (rowCount / partitionSize) + (rowCount % partitionSize == 0 ? 0 : 1);
+
+                // Generate SQL statements to read "pages" of data
+                for (int i = 0; i < numberOfFetches; i++) {
+                    FlowFile sqlFlowFile = null;
+                    try {
+                        Integer limit = partitionSize == 0 ? null : partitionSize;
+                        Integer offset = partitionSize == 0 ? null : i * partitionSize;
+                        final String query = dbAdapter.getSelectStatement(tableName, columnNames, StringUtils.join(maxValueClauses, " AND "), null, limit, offset);
+                        sqlFlowFile = session.create();
+                        sqlFlowFile = session.write(sqlFlowFile, out -> {
+                            out.write(query.getBytes());
+                        });
+                        session.transfer(sqlFlowFile, REL_SUCCESS);
+
+                    } catch (Exception e) {
+                        logger.error("Error while generating SQL statement", e);
+                        if (sqlFlowFile != null) {
+                            session.remove(sqlFlowFile);
+                        }
+                    }
+                }
+
+            } else {
+                // Something is very wrong here, one row (even if count is zero) should be returned
+                throw new SQLException("No rows returned from metadata query: " + selectQuery);
+            }
+
+
+            session.commit();
+            try {
+                // Update the state
+                stateManager.setState(statePropertyMap, Scope.CLUSTER);
+            } catch (IOException ioe) {
+                getLogger().error("{} failed to update State Manager, observed maximum values will not be recorded", new Object[]{this, ioe});
+            }
+        } catch (final ProcessException | SQLException e) {
```

Comment:
Does this try/catch need to be this large? It could fail after the sql query already ran successfully and fail later for an un-related reason and it would be prepended with "Unable to execute SQL select query {} due to {}"